src/main/resources/assets/computercraft/lua/rom/apis/pastebin.lua

```diff
@@ -0,0 +1,146 @@
+
+--- Attempts to guess the pastebin ID from the given code or URL
+local function extractId(paste)
+  local patterns = {
+      "^([%a%d]+)$",
+      "^https?://pastebin.com/([%a%d]+)$",
+      "^pastebin.com/([%a%d]+)$",
+      "^https?://pastebin.com/raw/([%a%d]+)$",
+      "^pastebin.com/raw/([%a%d]+)$",
+  }
+
+  for i = 1, #patterns do
+      local code = paste:match( patterns[i] )
+      if code then return code end
+  end
+
+  return nil
+end
+
+function download(url)
+  if not http then
+    return false, "Pastebin requires http API"
+  end
+
+  if type( url ) ~= "string" then
+    error( "bad argument #1 (expected string, got " .. type( url ) .. ")", 2 )
+  end
+
+  local paste = extractId( url )
+  if not paste then
+      return false, "Invalid pastebin code."
+  end
+
+  -- Add a cache buster so that spam protection is re-checked
+  local cacheBuster = ("%x"):format(math.random(0, 2^30))
+  local response, err = http.get(
+      "https://pastebin.com/raw/"..textutils.urlEncode( paste ).."?cb="..cacheBuster
+  )
+
+  if response then
```

Comment:
Invert this `if`, and return the error message (so `if not response then return nil, err end`). Just means flow is a little more linear :).